framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3559,6 +3559,27 @@ protected void checkEnclosingExpr(NewClassTree node, AnnotatedExecutableType con
                 argumentReceiverType = atypeFactory.getAnnotatedType(node.getEnclosingExpression());
             } else {
                 argumentReceiverType = atypeFactory.getReceiverType(node);
+                /*
+                  Consider the following code:
+                  Class Outer {
+                      Class Inner{}
+                  }
+                  Class Top{
+                      void test(Outer outer) {
+                          outer.new Inner(){};
+                      }
+                  }
+                  In Java versions below 11, the argumentReceiverType of outer.new Inner(){} is Top instead of Outer, because java8 organizes
```

Comment:
Should the `java8` be something else?
If it should be `Java 8`, what about Java 9 and 10?
Or should it be `Java below 11` to be consistent with the first part of this sentence?

Reply:
Thanks! Done in my local.